### PR TITLE
Add lint check gear imports another gear

### PR DESCRIPTION
**Pull request by kaojohnny**
connects #771

## note
- currently, skygear next imports server pkg for `skyerr`, `logging` and other utilities, so the lint ignores import `server` temporarily.

### CODE REVIEW

**Comment by kaojohnny** [comment]
@Steven-Chan Sorry, I forgot to add WIP to th title, I am adding it first to check travis work properly. I am updaing those features. Thanks.

**Comment by kaojohnny** [comment]
@Steven-Chan Updated.

File: tools/nextimportslint.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	tolerantImports := getTolerantImports(gearName, astFile.Comments)
+
+	for _, s := range astFile.Imports {
+		// s.Paht.Value contains `" at head and tail
+		cleanImportPath := strings.Replace(s.Path.Value, "\"", "", 2)
+		if strings.HasPrefix(cleanImportPath, gearBasePath) {
+			importGearRelativePath := strings.Replace(cleanImportPath, gearBasePath, "", 1)
+			importGearName := strings.Split(importGearRelativePath, "/")[0]
+			if illegalImport(gearName, importGearName, tolerantImports) {
+				errMsg := fmt.Sprintf("nextimports: doesn't allow \"%s\" imports aother gear \"%s\"", filename, importGearName)
```
Comment (Steven-Chan) [inline tools/nextimportslint.go]: `aother` -> `another`

Comment (Steven-Chan) [reply tools/nextimportslint.go]: `imports` -> `import`

Comment (kaojohnny) [reply tools/nextimportslint.go]: 😱

File: tools/nextimportslint.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	fset := token.NewFileSet() // positions are relative to fset
+	mode := parser.ImportsOnly + parser.ParseComments
+	astFile, err := parser.ParseFile(fset, filename, src, mode)
+	if err != nil {
+		return err
+	}
+	tolerantImports := getTolerantImports(gearName, astFile.Comments)
+
+	for _, s := range astFile.Imports {
+		// s.Paht.Value contains `" at head and tail
```
Comment (carmenlau) [inline tools/nextimportslint.go]: Overall looks good, just one minor typo. should be `s.Path.Value`

**Pull request merged**